web/tests/functional/store/test_proj/double_suppress.cpp

```diff
@@ -0,0 +1,6 @@
+void foo()
+{
+    // codechecker_suppress [deadcode.DeadStores] Same suppress comment twice.
+    // codechecker_suppress [deadcode.DeadStores] Same suppress comment twice.
```

Comment:
This is not a typical use case when the type and the checker name is the same. I agree that we need to test it too but we need to introduce more test cases which test the followin scenarios too:
```
  // codechecker_suppress [all] comment
  // codechecker_suppress [deadcode.DeadStores] comment
```
```
  // codechecker_suppress [all] comment
  // codechecker_confirmed [deadcode.DeadStores] comment
```
```
  // codechecker_suppress [all, deadcode.DeadStores] comment
```

For example on the second use case when I run the CodeChecker parse command I don't get an error message that multiple source code comments can be found for the same report but on the third use case I get this error. It should give an error in both cases.

Also now for every use cases we need to create a separte source files because it will give an error message on the first source code comment duplication. Maybe we can generate these source files on the fly. Simply creating a temp file, analyze and parse it. What do you think? We are doing the same for example at here: https://github.com/Ericsson/codechecker/blob/master/web/tests/functional/detection_status/test_detection_status.py#L82

Reply:
I fixed the implementation. New tests added to check these cases.